rllib/policy/view_requirement.py

```diff
@@ -28,90 +31,91 @@ class ViewRequirement:
-                `Policy.compute_actions_from_input_dict()`).
-            used_for_training: Whether the data will be used for
-                training. If False, the column will not be copied into the
-                final train batch.
+        shift_arr is infered from the shift value.
+
+        For example:
+            - if shift is -1, then shift_arr is np.array([-1]).
+            - if shift is [-1, -2], then shift_arr is np.array([-2, -1]).
+            - if shift is "-2:2", then shift_arr is np.array([-2, -1, 0, 1, 2]).
         """
-        self.data_col = data_col
-        self.space = (
-            space
-            if space is not None
-            else gym.spaces.Box(float("-inf"), float("inf"), shape=())
-        )
 
-        self.shift = shift
+        if self.space is None:
+            self.space = gym.spaces.Box(float("-inf"), float("inf"), shape=())
 
+        # TODO: ideally we won't need shift_from and shift_to, and shift_step.
+        # all of them should be captured within shift_arr.
         # Special case: Providing a (probably larger) range of indices, e.g.
         # "-100:0" (past 100 timesteps plus current one).
-        self.shift_from = self.shift_to = None
+        self.shift_from = self.shift_to = self.shift_step = None
         if isinstance(self.shift, str):
-            f, t = self.shift.split(":")
+            split = self.shift.split(":")
+            assert len(split) in [2, 3], f"Invalid shift str format: {self.shift}"
+            if len(split) == 2:
+                f, t = split
+                self.shift_step = 1
+            else:
+                f, t, s = split
+                self.shift_step = int(s)
+
             self.shift_from = int(f)
             self.shift_to = int(t)
 
-        self.index = index
-        self.batch_repeat_value = batch_repeat_value
-
-        self.used_for_compute_actions = used_for_compute_actions
-        self.used_for_training = used_for_training
-
-    def __str__(self):
-        """For easier inspection of view requirements."""
-        return "|".join(
-            [
-                str(v)
-                for v in [
-                    self.data_col,
-                    self.space,
-                    self.shift,
-                    self.shift_from,
-                    self.shift_to,
-                    self.index,
-                    self.batch_repeat_value,
-                    self.used_for_training,
-                    self.used_for_compute_actions,
-                ]
-            ]
-        )
+        shift = self.shift
+        self.shfit_arr = None
+        if self.shift_from:
+            self.shift_arr = np.arange(
+                self.shift_from, self.shift_to + 1, self.shift_step
+            )
+        else:
+            if isinstance(shift, int):
+                self.shift_arr = np.array([shift])
+            elif isinstance(shift, list):
+                self.shift_arr = np.array(shift)
```

Comment:
what if shift is [-5, -1, 1] though? basically the np array version of "-5:-1:1"

Reply:
what do you mean?
Do you mean "-5:-1"? that will not come to this if statement. 
These statements cover two cases: 1) shift = -1 which is line 113, 2) shift = [-3, -1, 1] indices of t-3, t-1, t+1 for each index t.